requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c

```diff
@@ -0,0 +1,51 @@
+/* Body file for function Controller
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "controller.h"
+
+#include <CbiPartitionApi.h>
+
+// Implemented in manager.c
+extern void print_rtems(const char *str);
+
+static void startup_on_processor_reset()
+{
+   print_rtems("Controller starting up on processor reset\n");
+}
+
+static void startup_on_power_reset()
+{
+   print_rtems("Controller starting up on power reset\n");
```

Comment:
Maybe "power reset" -> "power on", I think it would be cleaner

Reply:
Yes, power-on reset is the exact wording of the requirements, will change.